File: leetcode2/0203. Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        node=ListNode(0)
        node.next=head
        p=node
        while p and p.next:
            if p.next.val==val:
                p.next=p.next.next
            else:
                p=p.next
        return node.next

File: archives/C#/0203. Remove Linked List Elements.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode RemoveElements(ListNode head, int val) {
        ListNode node=new ListNode(0);
        node.next=head;
        ListNode p=node;
        while (p.next!=null){
            if (p.next.val==val){
                p.next=p.next.next;
            }
            else{
                p=p.next;
            }
        }
        return node.next;
    }
}
